src/xclient_zmq.cpp

```diff
@@ -0,0 +1,108 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xeus-zmq/xclient_zmq.hpp"
+#include "xclient_zmq_impl.hpp"
+
+namespace xeus
+{
+
+    xclient_zmq::xclient_zmq(std::unique_ptr<xclient_zmq_impl> impl)
+        : p_client_impl(std::move(impl))
+    {
+    }
+
+    // Has to be in the cpp because incomplete
+    // types are used in unique_ptr in the header
+    xclient_zmq::~xclient_zmq() = default;
+
+
+    void xclient_zmq::send_on_shell(xmessage msg)
+    {
+        p_client_impl->send_on_shell(std::move(msg));
+    }
+
+    void xclient_zmq::send_on_control(xmessage msg)
+    {
+        p_client_impl->send_on_control(std::move(msg));
+    }
+
+    std::optional<xmessage> xclient_zmq::check_shell_answer()
+    {
+        return p_client_impl->receive_on_shell(-1);
```

Comment:
So `check_shell_answer` doesn't have a `timeout` parameter but `receive_on_shell` does as I think as we want to wait indefinitely ... I ended up passing -1 here. Is that how we should do it ?

Reply:
Yep that's the way to do it.